Modules/IndividualAssessment/classes/Settings/class.ilIndividualAssessmentSettings.php

```diff
@@ -113,34 +113,58 @@ public function isResultVisible(): bool
+                               ->withValue($this->isFileRequired());
+        $file_visible = $input->checkbox($lng->txt("iass_file_visible_examinee"), '')
+                              ->withValue($this->isFileVisible());
+        $notification = $input->checkbox($lng->txt("iass_notify"), $lng->txt("iass_notify_explanation"))
+                              ->withValue($this->isResultVisible());
+
+        $fields = [
+            "title" => $title,
+            "description" => $description,
+            "content" => $content
+        ];
+
+        if (!$specified_form) {
+            $fields['record_template'] = $record_template;
+            $fields['event_time_place_required'] = $time_place_required;
+            $fields['file_required'] = $file_required;
+            $fields['file_visible'] = $file_visible;
+        }
+
+        $fields['result_visible'] = $notification;
+
         return $input->section(
-            [
-                $input->text($lng->txt("title"))
-                    ->withValue($this->getTitle())
-                    ->withRequired(true),
-                $input->textarea($lng->txt("description"))
-                    ->withValue($this->getDescription()),
-                $input->textarea($lng->txt("iass_content"), $lng->txt("iass_content_explanation"))
-                    ->withValue($this->getContent()),
-                $input->textarea($lng->txt("iass_record_template"), $lng->txt("iass_record_template_explanation"))
-                    ->withValue($this->getRecordTemplate()),
-                $input->checkbox($lng->txt("iass_event_time_place_required"), $lng->txt("iass_event_time_place_required_info"))
-                    ->withValue($this->isEventTimePlaceRequired()),
-                $input->checkbox($lng->txt("iass_file_required"), $lng->txt("iass_file_required_info"))
-                    ->withValue($this->isFileRequired()),
-                $input->checkbox($lng->txt("iass_file_visible_examinee"), '')
-                    ->withValue($this->isFileVisible()),
-                $input->checkbox($lng->txt("iass_notify"), $lng->txt("iass_notify_explanation"))
-                    ->withValue($this->isResultVisible()),
-            ],
+            $fields,
             $lng->txt("settings")
         )->withAdditionalTransformation(
             $refinery->custom()->transformation(function ($value) {
                 return new ilIndividualAssessmentSettings(
                     $this->getObjId(),
-                    ...$value
+                    $value['title'],
+                    $value['description'],
+                    $value['content'],
+                    $value['record_template'] ?? '',
+                    $value['event_time_place_required'] ?? false,
+                    $value['file_required'] ?? false,
+                    $value['file_visible'] ?? false,
+                    $value['result_visible']
```

Comment:
das ist irgendwie nicht schöner geworden :)
Analog zu den changes in ilIndividualAssessmentUserGrading denke ich, dass wir mit optionalen Parametern an der Klasse mehr erreichen, sowohl in Bedienung als auch an Transparenz.